### PR TITLE
Avoid re-building xtask in dev server & various fixes

### PR DESCRIPTION
**Pull request by cecton**

### CODE REVIEW

File: examples/demo/xtask/src/main.rs [hunk 1]
```diff
@@ -50,14 +47,14 @@ fn main() -> Result<()> {
...
             arg.run(build_command)?;
         }
-        Command::Serve(arg) => {
-            log::info!("Starting to serve");
-            build_command.arg("--optimize");
-            arg.command(build_command)
-                .start(xtask_wasm::default_dist_dir(false))?;
+        Command::Start(arg) => {
+            log::info!("Starting to development server...");
+            arg.arg("dist").start(xtask_wasm::default_dist_dir(false))?;
```
Comment (yozhgoor) [inline examples/demo/xtask/src/main.rs:57]: Can you use this in the `DevServer` example too?

src/dev_server.rs line 43-49

**Review by yozhgoor — CHANGES_REQUESTED** [review]
Thanks for the PR :pray:

File: examples/demo/xtask/src/main.rs [hunk 2]
```diff
@@ -50,14 +47,14 @@ fn main() -> Result<()> {
             }
         }
         Command::Watch(arg) => {
-            log::info!("Starting to watch");
+            log::info!("Watching for changes and check...");
+            let mut build_command = process::Command::new("cargo");
+            build_command.arg("check");
             arg.run(build_command)?;
```
Comment (cecton) [inline examples/demo/xtask/src/main.rs:53]:
```suggestion
            log::info!("Watching for changes and check...");
            let mut command = process::Command::new("cargo");
            command.arg("check");
            arg.run(command)?;
```

File: examples/demo/xtask/src/main.rs (hunk 1 above)
Comment (cecton) [reply examples/demo/xtask/src/main.rs:57]: Good catch!

**Review by yozhgoor — APPROVED** [review]

**Pull request merged**